src/Chartjs.vue

```diff
@@ -38,19 +38,32 @@ export default {
     })
   },
 
-  data () {
-    return {
-      chart: null
+  data () { return {
+    chart: null
+  }},
+
+  methods: {
+    resetChart() {
+      this.$nextTick(() => {
+        this.chart.destroy()
+        this.chart = new Chart(this.$el, {
+          type: this.type,
+          data: this.data,
+          options: this.options
+        })
+      })
     }
   },
 
   watch: {
+    type (val) {
+      this.resetChart()
+    },
     data (val) {
-      this.$nextTick(() => {
-        // this.chart.data.datasets = val.datasets
-        // this.chart.data.labels = val.labels
-        this.chart.update()
-      })
+      this.resetChart()
```

Comment:
I think here we need `this.chart.update()`

and if `val` is useless, please remove it.